worker/instancemutater/doc.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+
+// Package instancemutater defines workers that checks the list of lxd profiles
+// applied to a machine against the list of expected profiles based on the
+// application version which should be running on the machine. In particular, it
+// creates two workers from the same code with different configurations; the
+// ContainerWorker, and the EnvironWorker.
+//
+// The ContainerWorker runs on a machine and watches for containers to be
+// created on it.
+//
+//      ┌───────────────────────────────┐
+//      │       MACHINE                 │
+//      │                               │
+//      │                               │
+//      │   ┌──────────────────────┐    │
+//      │   │                      │    │
+//      │   │     CONTAINER        │    │
+// ┌────┼───►                      │    │
+// │    │   │                      │    │
+// │    │   │  ┌────────────────┐  │    │
+// │    │   │  │   UNIT         │  │    │
+// │    │   │  │                │  │    │
+// │    │   │  │                │  │    │
+// │    │   │  │ ┌────────────┐ │  │    │
+// │    │   │  │ │ CHARM      │ │  │    │
+// │    │   │  │ │            │ │  │    │
+// │    │   │  │ └─────┬──────┘ │  │    │
+// │    │   │  │       │        │  │    │
+// │    │   │  └───────┼────────┘  │    │
+// │    │   │          │           │    │
+// │    │   └──────────┼───────────┘    │
+// │    │              │                │
+// │    └──────────────┼────────────────┘
+// │                   │
+// └───────────────────┘
+//      LXD PROFILE
+//
+// The EnvironWorker watches for machines in the model to be created.
+// 
+//      ┌───────────────────────────────┐
+//      │       MACHINE                 │
+//      │                               │
+// ┌────►                               │
+// │    │   ┌──────────────────────┐    │
+// │    │   │                      │    │
+// │    │   │     CONTAINER        │    │
+// │    │   │                      │    │
+// │    │   │                      │    │
+// │    │   │  ┌────────────────┐  │    │
+// │    │   │  │   UNIT         │  │    │
+// │    │   │  │                │  │    │
+// │    │   │  │                │  │    │
+// │    │   │  │ ┌────────────┐ │  │    │
+// │    │   │  │ │ CHARM      │ │  │    │
+// │    │   │  │ │            │ │  │    │
+// │    │   │  │ └─────┬──────┘ │  │    │
+// │    │   │  │       │        │  │    │
+// │    │   │  └───────┼────────┘  │    │
+// │    │   │          │           │    │
+// │    │   └──────────┼───────────┘    │
+// │    │              │                │
+// │    └──────────────┼────────────────┘
+// │                   │
+// └───────────────────┘
+// LXD PROFILE
+//
+// To understand this better with a similar mechanism, take a look at the
+// provisioner worker as well.
```

Comment:
Is it important to say that they are created from the same code with different configurations?

Suggestion: Don't mention the names of the workers in-line -- just start a list and then define each one of them there directly.  It would also help if the descriptions were more parallel. E.g., can we say that the second worker "runs on a model and..."? 

Example rewrite:

> Package instancemutater defines two workers that check the list of LXD profiles based on the application version which should be running on the machine, as follows:
    - ContainerWorker: Runs on a machine and watches for any containers being created on it.
    - EnvironWorker: Runs on a model and watches for any machines being created on it. 

Clarification questions: 

(1) What does this mean: "checks the list of ...  based on...". What does the check really do?
(2) What's the connection between checking lxd profiles and watching for containers / machines being created on...?
(3) In "application version which should be running on the machine", does "machine" really mean "container or machine", as suggested by the description of the two workers and consistently with https://juju.is/docs/juju/machine#heading--machines-and-system-lxd-containers ?

Reply:
@tmihoc From a internal juju perspective a "machine" and a "container" are a "machine".